pagopa/pn-frontend-e2e
# PN-9242-e-2-e-mittente-invio-notifica-a-destinatario-con-indirizzo-fornito-dalla-pa

## Short description
There isn't a test related to the **PN-9242** as you can see in the [confluence page](https://pagopa.atlassian.net/wiki/spaces/PN/pages/868778282/Mittente+-+Mapping+test-scenari)

## List of changes proposed in this pull request
- added new test file `128_A1_invioNotificaADestinatarioConIndirizzoFornitoDallaPA.feature`
- added new step file class `WebTool` with new utility methods:
```
- switchToPortal()
- openNewTab()
- closeTab()
```
- moved methods: `generatePaProtocolNumber(), decodeNotificationRequestId() and waitTime()` into `WebTool`
- added new feature steps into `NotificaMittentePagoPATest` such as: 
```
- Then Nella section Informazioni preliminari si inseriscono i dati della notifica
- Then Nella section Destinatario si inseriscono i dati del destinatario
- And Nella section Destinitario si clicca su {string} e si inseriscono i dati
- Then Nella section Allegati si carica un atto
- And Si verifica che la notifica è stata creata correttamente
- Then In parallelo si effettua l'accesso al portale destinatario persona fisica e si apre la notifica ricevuta
- Si verifica che la notifica abbia lo stato {string}
```
- added related methods to the new feature steps
- moved `utility` package into the main project folder
- added 2 new enum that manage the login with the token


## How to test
Simply run `mvn clean test -Dcucumber.filter.tags=@TA_invioNotificaConIndirizzoFornitoDallaPA` and verify if the test works fine.
